### PR TITLE
added integration tests.

### PR DESCRIPTION
**Pull request by Mmalumbo**
Signed-off-by: mmalumbo <mmalumbo@gmail.com>

### CODE REVIEW

File: src/tests/failqueue.test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    await worker.destroy();
+    await server.close();
+  });
+
+  it('should successfully send data elements that failed migration to dhis2', async () => {
+    server = await fakeDhis2Server(3000, modes.succeed);
+    await migrate(config, connection, worker, message);
+    testMigrationDataElement = await connection.getRepository(MigrationDataElements).findOne(testMigrationDataElement);
+
+    expect(testMigrationDataElement.createdAt).toBeTruthy();
```
Comment (liwoo) [inline src/tests/failqueue.test.ts]: assertions should be more explicit, like assert against an actual date time

File: src/tests/fixtures/database.fixture.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import 'reflect-metadata';
+
+export const malu = 'maliu';
```
Comment (liwoo) [inline src/tests/fixtures/database.fixture.ts]: don't understand this...

File: src/tests/failqueue.test.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+  it('should failt to send data elements that failed migration to dhis2', async () => {
```
Comment (liwoo) [inline src/tests/failqueue.test.ts]: we need to find a way of testing the number of retries...

**Review by liwoo — CHANGES_REQUESTED** [review]

File: src/tests/failqueue.test.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import { createWorker, Message } from '../migration-failure-worker';
+
+import { loadConfig } from '../config';
+import { connectToDatabase } from '../datasource';
+import { migration, modes, fakeDhis2Server, migrationDataElements } from './fixtures';
+import { Migration } from '../models/Migration';
+import { MigrationDataElements } from '../models/MigrationDataElements';
+import { migrate } from '../migration';
+
+let path: string;
```
[the remark this answers is not in the document]
Comment (Mmalumbo) [reply src/tests/failqueue.test.ts]: changed it to const.

File: src/tests/failqueue.test.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+let config: DotenvParseOutput;
+let connection: Connection;
+let testMigration: Migration;
+let testMigrationDataElement: MigrationDataElements;
+let worker: Worker;
+let message: Message;
+let server: any;
+
+describe('', () => {
```
[the remark this answers is not in the document]
Comment (Mmalumbo) [reply src/tests/failqueue.test.ts]: added a name.

**Comment by Mmalumbo** [comment]
ooo

**Pull request closed without merging**